hbase-server/src/main/java/org/apache/hadoop/hbase/coprocessor/MultiRowMutationEndpoint.java

```diff
@@ -221,22 +223,25 @@ private boolean matches(Region region, ClientProtos.Condition condition) throws
       get.setTimeRange(timeRange.getMin(), timeRange.getMax());
     }
 
-    List<Cell> result = region.get(get, false);
     boolean matches = false;
-    if (filter != null) {
-      if (!result.isEmpty()) {
-        matches = true;
-      }
-    } else {
-      boolean valueIsNull = comparator.getValue() == null || comparator.getValue().length == 0;
-      if (result.isEmpty() && valueIsNull) {
-        matches = true;
-      } else if (result.size() > 0 && result.get(0).getValueLength() == 0 && valueIsNull) {
-        matches = true;
-      } else if (result.size() == 1 && !valueIsNull) {
-        Cell kv = result.get(0);
-        int compareResult = PrivateCellUtil.compareValue(kv, comparator);
-        matches = matches(op, compareResult);
+    try (RegionScanner scanner = region.getScanner(new Scan(get))) {
+      List<Cell> result = new ArrayList<>();
```

Comment:
nit:  a developer seeing this code w/o the proper background might look at this code block and think -- why scan to do one next ? Why not just replace it w/ a 'get'? Does this need a comment? If only to point at the JIRA?

Reply:
I think we'd better find a way to fix the get method itself...

Reply:
Thanks @saintstack @Apache9 , I'll change the HRegion.get() to make a heap copy of the results use it in the CPs, while in the checkAndMutate method, I'll add comments for the background and why not using a get().

Reply:
excellent